src/mds/Server.cc

```diff
@@ -580,6 +580,13 @@ void Server::handle_client_session(const cref_t<MClientSession> &m)
   uint64_t sseq = 0;
   switch (m->get_op()) {
   case CEPH_SESSION_REQUEST_OPEN:
+    if(mds->mdsmap->get_reject_client_session()) {
+      dout(0) << "cannot accept session when recovery in progress" << dendl;
+      auto reply = make_message<MClientSession>(CEPH_SESSION_REJECT);
+      reply->metadata["error_string"] = "session reject due to recovery in progress";
```

Comment:
ditto

Reply:
Also, make the client smarter to not connect to an MDS until the flag is cleared. Output something regularly that connections cannot be made while the flag is set.

Reply:
like running a thread sending `SESSION_REJECT` to client untill flag is unset?

Reply:
No. I mean in `src/client/Client.cc`. It should not try to mount if it sees that flag set in the `MDSMap`.

Reply:
i had that on my mind, but what if there is older client, it would not work in that case right?

Reply:
but yeah that MDS is anyways going to handle it in that case right. Cool, I'll add it in Client

Reply:
> i had that on my mind, but what if there is older client, it would not work in that case right?

No, but that's fine.

Reply:
@dparmar18 This is what we had talked before.

Reply:
@lxbsz yeah, I do remember, thanks for the suggestion! BTW I have pushed the changes, PTAL